docs/design/database.md

```diff
@@ -25,61 +25,77 @@
 
 # Table design
 
-Historical state changes are indexed by autoincrementing enumeration called `TransitionId`. This means that `reth` stores the state for every account after every transaction that touched it, and it provides indexes for accessing that data quickly. While this may make the database size bigger (needs benchmark once `reth` is closer to prod), it also enables blazing-fast historical tracing and simulations because there is no need to re-execute all transactions inside a block.
-
-State transitions differ from transaction level enumeration in order to account for block rewards.
-e.g. If block #1 contains transactions from id 0 to 5 (inclusive) and has a reward, the reward will be assigned transition id 6 and transition ids in block #2 will start block index 7.
-
-Two additional tables `TxTransitionIdIndex` and `BlockTransitionIdIndex` are introduced to enable reverse lookup of transition by transaction id and block number respectively. The transition id at block number denotes the final state transition of the block. If block has a reward, the state transition at block number `x` points at data about the state transition for the reward distribution. Otherwise, it points to the state transition at the last transaction in block number `x`.
+Historical state changes are indexed by `BlockNumber`. This means that `reth` stores the state for every account after every block that touched it, and it provides indexes for accessing that data quickly. While this may make the database size bigger (needs benchmark once `reth` is closer to prod).
 
 Below, you can see the table design that implements this scheme:
 
 ```mermaid
 erDiagram
+CanonicalHeaders {
+    u64 BlockNumber "PK"
+    H256 HeaderHash "Value for CanonicalHeaders"
+}
+HeaderNumbers {
+    H256 BlockHash "PK"
+    u64 BlockNumber
+}
+Headers {
+    u64 BlockNumber "PK"
+    Header Data
+}
+BlockBodyIndices {
+    u64 BlockNumber "PK"
+    u64 first_tx_num
+    u64 tx_count
+}
+Receipts {
+    u64 TxNumber "PK"
+    Receipt Data
+}
 Transactions {
     u64 TxNumber "PK"
-    Transaction Data
+    TransactionSigned Data
```

Comment:
this was just changed in https://github.com/paradigmxyz/reth/pull/2279 
this is now `TransactionSignedNoHash`
```suggestion
    TransactionSignedNoHash Data
```